server/src/main/java/com/linecorp/centraldogma/server/CorsConfig.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.server;
+
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Streams;
+
+/**
+ * CORS configuration.
+ */
+public final class CorsConfig {
+
+    private static final int DEFAULT_MAX_AGE = 7200;
+
+    private final List<String> allowedOrigins;
+    private final int maxAge;
+
+    /**
+     * Creates an instance with the specified {@code allowedOrigins} and
+     * {@code maxAge}.
+     */
+    @JsonCreator
+    public CorsConfig(@JsonProperty("allowedOrigins") Object allowedOrigins,
+                      @JsonProperty("maxAge") @Nullable Integer maxAge) {
+        if (allowedOrigins instanceof Iterable &&
+            Streams.stream((Iterable<?>) allowedOrigins).allMatch(String.class::isInstance)) {
+            this.allowedOrigins = Lists.newArrayList((Iterable<String>) allowedOrigins);
+        } else if (allowedOrigins instanceof String) {
+            final String origin = (String) allowedOrigins;
+            this.allowedOrigins = Collections.singletonList(origin);
+        } else {
+            throw new IllegalArgumentException(
+                    "allowedOrigins: " + allowedOrigins +
+                    " (expected: either a string or an array of strings)");
+        }
+        if (this.allowedOrigins.isEmpty()) {
+            throw new IllegalArgumentException(
+                    "allowedOrigins: " + allowedOrigins +
+                    " (expected: the list of origins must not be empty)");
+        }
+
+        this.maxAge = maxAge == null || maxAge <= 0 ? DEFAULT_MAX_AGE : maxAge;
```

Comment:
`-1` or `0` could mean an unlimited age. If we don't provide the unlimited age, I prefer disallowing `0` and a negative number and throwing `IllegalArgumentException`.
What do you think of using the default value if `maxAge` is null?

Reply:
After searching a bit, cache can not be unlimited since the browser caches per requests, not the entire domain. If cache is unlimited, browser would run out of RAM. I also could not find a reference to when to set cache to 0 for CORS. To conclude, i will disallow <=0 values and provide set the default cache time (7200s) if user does not set (aka maxAge is null).